2022/Day7/RFC.md

```diff
@@ -0,0 +1,87 @@
+# AoC 2022 Day 6
+## Overview:
+You can hear birds chirping and raindrops hitting leaves as the expedition proceeds. Occasionally, you can even hear 
+much louder sounds in the distance; how big do the animals get out here, anyway?
+
+The device the Elves gave you has problems with more than just its communication system. You try to run a system update:
+
+```
+$ system-update --please --pretty-please-with-sugar-on-top
+Error: No space left on device
+```
+
+Perhaps you can delete some files to make space for the update?
+
+The filesystem consists of a tree of files (plain data) and directories (which can contain other directories or files). 
+The outermost directory is called /. You can navigate around the filesystem, moving into or out of directories and 
+listing the contents of the directory you're currently in.
+
+Since the disk is full, your first step should probably be to find directories that are good candidates for deletion. 
+To do this, you need to determine the total size of each directory. The total size of a directory is the sum of the 
+sizes of the files it contains, directly or indirectly. (Directories themselves do not count as having any intrinsic size.)
+
+## Glossary:
+
+## Part 2:
+### Goal:
+The total disk space available to the filesystem is 70000000. To run the update, you need unused space of at least 30000000. 
+You need to find a directory you can delete that will free up enough space to run the update.
+
+Find the smallest directory that, if deleted, would free up enough space on the filesystem to run the update. 
+What is the total size of that directory?
+
+### Design:
+#### Parsing:
+No additional parsing needed.
+
+#### Solution:
+The first thing to do is to find how much space we need to free up to get the requirement. We can do that by summing all
```

Comment:
How are you planning to sum all the directories?  The way this is worded makes me concerned you're going to get the answer wrong.

Reply:
I was going to use the `sum_all_subdirectories_under_limit` method that is currently there and set the limit to 70000000, since we know each individual directory is under that limit.

Reply:
But... that will get you a number much higher than 70000000.  Consider the structure:
```
/
-> dir a
   -> dir b
      -> dir c
         -> a.txt 5000
```

The size of dirs `/`, `a`, `b`, and `c` is 5000.  There's only 5000 being used.  But `sum_all_directories_under_limit(70000000)` would return 20000.

Reply:
Oh. Doh - I just need to get the size of "/" - which I have. And subtract that from 70000000. That's me not clearly reading things. Sorry. Will fix that